generator/src/main.rs

```diff
@@ -1517,9 +1518,10 @@ impl Parser {
                         match crate::fixed_str_bytes(&ext.extension_name) {
                             #(#ext_set_inits)*
                             bytes => {
-                                if let Ok(name) = std::str::from_utf8(bytes) {
-                                    out.other.push(name.into());
-                                }
+                                let cstr = CStr::from_bytes_with_nul(bytes)
+                                    .expect("extension names should be null terminated strings");
+                                let string = String::from_utf8_lossy(cstr.to_bytes()).to_string();
```

Comment:
I'm not sure it's a good idea to ever silently transform an extension name, since needing to do so again indicates a serious bug, and would break any comparisons. `CStr` also has some nice convenience methods for string conversion.
```suggestion
                                let string = cstr.to_str().expect("extension names should be valid UTF-8").to_string();
```